src/spect_experimental/utilities/ctac_to_mu_values.cxx

```diff
@@ -0,0 +1,224 @@
+    {
+      std::cerr << usage;
+      return EXIT_FAILURE;
+    }
+
+  //Read slope file
+  std::ifstream slope_json_file_stream(slope_filename);
+  nlohmann::json slope_json;
+  slope_json_file_stream >> slope_json;
+
+  //Put user-specified manufacturer into upper case.
+  std::string manufacturer = manufacturer_name;
+  std::locale loc;
+
+  for (std::string::size_type i=0; i<manufacturer.length(); ++i)
+    manufacturer[i] = std::toupper(manufacturer[i],loc);
+
+  stir::info(boost::format("Manufacturer: '%s'") % manufacturer);
+
+  //Get desired keV as integer value
+  int keV;
+
+  std::stringstream ss;
+  ss << keV_str;
+  ss >> keV;
+
+  stir::info(boost::format("target keV: '%i'") % keV);
+
+  //Extract appropriate chunk of JSON file for given manufacturer.
+  nlohmann::json target = slope_json["scale"][manufacturer]["transform"];
+
+  int location = -1;
+  int pos = 0;
+  for (auto entry : target){
+    if (entry["kev"] == keV)
+      location = pos;
+    pos++;
+  }
+
+  if (location == -1){
+    std::cerr << "Desired keV: " << keV << " not found! ";
+    std::cerr << "Aborting!";
+    return EXIT_FAILURE;
+  }
+
+  //Extract transform for specific keV.
+  nlohmann::json j = target[location];
+  //std::cout << j.dump(4);
+
+  //Read DICOM data
+  stir::info(boost::format("ctac_to_mu_values: opening file %1%") % input_filename);
+  std::unique_ptr< FloatImageType > input_image_sptr(stir::read_from_file<FloatImageType>( input_filename ));
+
+  unique_ptr<VoxelsOnCartesianGrid<float> >image_aptr
+      (dynamic_cast<VoxelsOnCartesianGrid<float> *>(
+           DiscretisedDensity<3,float>::read_from_file(input_filename))
+      );
+
+  BasicCoordinate<3,int> min_indices, max_indices;
+  if (!image_aptr->get_regular_range(min_indices, max_indices))
+    error("Non-regular range of coordinates. That's strange.\n");
+
+  BasicCoordinate<3,float> edge_min_indices(min_indices), edge_max_indices(max_indices);
+  edge_min_indices-= 0.5F;
+  edge_max_indices+= 0.5F;
+
+  std::cout << "\nOrigin in mm {z,y,x}    :" << image_aptr->get_origin()
+            << "\nVoxel-size in mm {z,y,x}:" << image_aptr->get_voxel_size()
+            << "\nMin_indices {z,y,x}     :" << min_indices
+            << "\nMax_indices {z,y,x}     :" << max_indices
+            << "\nNumber of voxels {z,y,x}:" << max_indices - min_indices + 1
+            << "\nPhysical coordinate of first index in mm {z,y,x} :"
+            << image_aptr->get_physical_coordinates_for_indices(min_indices)
+            << "\nPhysical coordinate of last index in mm {z,y,x}  :"
+            << image_aptr->get_physical_coordinates_for_indices(max_indices)
+            << "\nPhysical coordinate of first edge in mm {z,y,x} :"
+            << image_aptr->get_physical_coordinates_for_indices(edge_min_indices)
+            << "\nPhysical coordinate of last edge in mm {z,y,x}  :"
+            << image_aptr->get_physical_coordinates_for_indices(edge_max_indices);
+  std::cout << std::endl;
```

Comment:
delete this. it creates a dependency on `VoxelsOnCartesianGrid that we don't really need.

Reply:
Done 0061adf.